test/System.Net.Http.Formatting.Test/Formatting/JsonMediaTypeFormatterTests.cs

```diff
@@ -576,6 +577,29 @@ public override Task WriteToStreamAsync_UsesCorrectCharacterEncoding(string cont
                 formatter, content, formattedContent, mediaType, encoding, isDefaultEncoding);
         }
 
+#if NET6_0_OR_GREATER
+        // Cannot Mock a Stream and let JsonWriter write to it. Writer will use ReadOnlySpan in this case and such
+        // parameters are not currently mockable. See moq/moq4#829, moq/moq4#979, and dotnet/runtime#45152.
+        // Override here avoids the Mock<Stream> and should confirm this Stream is not closed. Also adds an
+        // additional check of the written text.
```

Comment:
@stephentoub this relates to one of our email conversations. Thanks for your pointer to dotnet/runtime#45152 and helping me understand what's happening.

Thoughts from all reviewers much appreciated: Should I simply replace [MediaTypeFormatterTestBase.WriteToStreamAsync_WhenObjectIsNull_WritesDataButDoesNotCloseStream](https://github.com/aspnet/AspNetWebStack/blob/main/test/System.Net.Http.Formatting.Test/Formatting/MediaTypeFormatterTestBase.cs#L205) along these lines❔ Would need a `protected` property containing the expected `Stream` content for a `null` value. But I'm thinking verifying what's written makes sense. I'm also thinking the BSON case might be more difficult to confirm.